### PR TITLE
opendatahub-io/vllm: add IBM main branch build

### PR DESCRIPTION
**Pull request by z103cb**
Added builds for the IBM main branch.

### CODE REVIEW

File: ci-operator/config/opendatahub-io/vllm/opendatahub-io-vllm-main.yaml [hunk 1]
```diff
@@ -9,6 +9,8 @@ images:
     value: "1"
   - name: BUILDKIT_PROGRESS
     value: plain
+  - name: max_jobs
```
Comment (dtrifiro) [inline ci-operator/config/opendatahub-io/vllm/opendatahub-io-vllm-main.yaml:12]: Might need to reduce the max jobs or increase memory, ci is failing with:

```console
  error occurred handling build vllm-main-cuda-intel-gpu-amd64: build not successful after 5 attempts: [the build vllm-main-cuda-intel-gpu-amd64 failed after 37m31s with reason BuildPodEvicted: The node was low on resource: memory. Threshold quantity: 100Mi, available: 91096Ki. Container docker-build was using 25077488Ki, request is 3885454545, has larger consumption of memory. , the build vllm-main-cuda-intel-gpu-amd64 failed after 39m29s with reason BuildPodEvicted: The node was low on resource: memory. Threshold quantity: 100Mi, available: 74172Ki. Container docker-build was using 27370076Ki, request is 3885454545, has larger consumption of memory. 

Error on reading termination message from logs: failed to ...-b605cf28602e/docker-build/0.log: no such file or directory, the build vllm-main-cuda-intel-gpu-amd64 failed after 8m39s with reason BuildPodEvicted: The node was low on resource: memory. Threshold quantity: 100Mi, available: 27056Ki. Container docker-build was using 6712380Ki, request is 3885454545, has larger consumption of memory. , the build vllm-main-cuda-intel-gpu-amd64 failed after 42m13s with reason BuildPodEvicted: The node was low on resource: memory. Threshold quantity: 100Mi, available: 71932Ki. Container docker-build was using 27122188Ki, request is 3966666666, has larger consumption of memory. , the build vllm-main-cuda-intel-gpu-amd64 failed after 50m17s with reason BuildPodEvicted: The node was low on resource: memory. Threshold quantity: 100Mi, available: 94660Ki. Container docker-build was using 27177320Ki, request is 3966666666, has larger consumption of memory. ]}
```

**Comment by bearnl** [comment]
/pj-rehearse periodic-ci-opendatahub-io-vllm-ibm_main-nightly-image-mirror

File: ci-operator/config/opendatahub-io/vllm/opendatahub-io-vllm-ibm_main.yaml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    dependencies:
+      SOURCE_IMAGE_REF: vllm-main-cuda-intel-gpu
+    env:
+      IMAGE_REPO: vllm
+    workflow: opendatahub-io-ci-image-mirror
+- as: fast-image-mirror
+  postsubmit: true
+  steps:
+    dependencies:
+      SOURCE_IMAGE_REF: vllm-main-cuda-intel-gpu
```
Comment (Xaenalt) [inline ci-operator/config/opendatahub-io/vllm/opendatahub-io-vllm-ibm_main.yaml:67]: Change name here as well

File: ci-operator/config/opendatahub-io/vllm/opendatahub-io-vllm-ibm_main.yaml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      cpu: "16"
+      memory: 24Gi
+    requests:
+      cpu: "16"
+      memory: 24Gi
+tests:
+- as: pr-image-mirror
+  steps:
+    dependencies:
+      SOURCE_IMAGE_REF: vllm-main-cuda-intel-gpu
```
Comment (Xaenalt) [inline ci-operator/config/opendatahub-io/vllm/opendatahub-io-vllm-ibm_main.yaml:59]: Change name here as well

File: ci-operator/config/opendatahub-io/vllm/opendatahub-io-vllm-ibm_main.yaml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    env:
+      IMAGE_REPO: vllm
+      RELEASE_VERSION: fast-ibm
+    workflow: opendatahub-io-ci-image-mirror
+- as: nightly-image-mirror
+  cron: 0 0 * * *
+  steps:
+    dependencies:
+      SOURCE_IMAGE_REF: vllm-main-cuda-intel-gpu
```
Comment (Xaenalt) [inline ci-operator/config/opendatahub-io/vllm/opendatahub-io-vllm-ibm_main.yaml:76]: Change name here as well

**Pull request merged**